Add new function: skip vlan, scan vlan, and scan trunk port

xxconn opened the pull request:
Add new function:

In site, add skip vlan, and scan vlan

In device, add scan-trunk-port for VM.

xxconn committed:
Update poller_mactrack.php

Formating the source code

xxconn committed:
Update mactrack_functions.php

Formatting the code

xxconn committed:
Update mactrack_functions.php

Formatting the code

xxconn committed:
Update mactrack_view_macs.php

1. Fix  the IP address's delimiter.
2. Formatting the code.

xxconn committed:
Update mactrack_macauth.php

Formatting the code

xxconn committed:
Update mactrack_macauth.php

Formatting the code

xxconn committed:
Merge branch 'Cacti:develop' into develop

xxconn committed:
Update mactrack_view_arp.php

Adjust the text filed with the left alignment for easy readable.

xxconn committed:
Update mactrack_macauth.php

The default to view the authorized mac address with descending sort is more useful.

xxconn committed:
Update mactrack_view_macs.php

1. Adjust the text filed with the left alignment for easy readable.

xxconn committed:
Update mactrack_cisco.php

The Cisco default vlan are 1002-1005 (4 vlans).

xxconn committed:
Update mactrack_cisco.php

In function get_IOS_dot1dTpFdbEntry_ports, it doesn't need to run collect ifIndexes information twice.

xxconn committed:
Merge branch 'Cacti:develop' into develop

xxconn committed:
Update mactrack_functions.php

Add ifType '53' for vlan trunk  and '161' for port-channel port

xxconn committed:
Update mactrack_actions.php

Add scan_vlan and skip_vlan for site, scan_trunk_port for device.

xxconn committed:
Update setup.php

Add skip_vlan and scan_vlan for site, scan_trunk_port for device.

xxconn committed:
Update mactrack_sites.php

Add skip_vlan and scan_vlan for site.

xxconn committed:
Update mactrack_devices.php

Add scan_trunk_port for device and manual choose for device type.

xxconn committed:
Update mactrack_cisco.php

Add skip_vlan and scan_vlan for device.

TheWitness commented on setup.php:
```diff
@@ -1464,6 +1485,26 @@ function mactrack_config_form () {
+		'class' => 'textAreaNotes',
+		'textarea_rows' => '5',
+		'textarea_cols' => '80',
+		'description' => 'VLAN ID always forced to be skipped scanning',
```
Should indicate is column separated or space in descriptions.  I think in MakTrack, it's generally comma. Double check though.

TheWitness commented on setup.php:
```diff
@@ -1322,6 +1332,17 @@ function mactrack_config_form () {
+	'scan_trunk_port' => array(
+		'method' => 'textarea',
+		'friendly_name' => 'Trunk Ports to Scan',
+		'description' => 'Some VMware network port will be configured with 802.1Q',
```
Same comment here.  You could say something like "A comma delimited is of ..."

TheWitness commented on setup.php:
```diff
@@ -1464,6 +1485,26 @@ function mactrack_config_form () {
+		'class' => 'textAreaNotes',
+		'textarea_rows' => '5',
+		'textarea_cols' => '80',
+		'description' => 'VLAN ID to be scaned',
```
Ditto on previous two comments.

xxconn replied:
Modified the new field's description.

TheWitness reviewed (changes requested):
Generally column additions require a version increase.  Just increment the point release in the information file.

xxconn committed:
Update setup.php

Modify the new field's description.

xxconn commented:
I will modify INFO and add new fields to database.php tomorrow.

xxconn committed:
Update database.php

Add new database columns `skip_vlans`, `scan_vlans`, `scan_trunk_port`

xxconn committed:
Update INFO

1. Add some new features like vlans to skip or scan, and trunk port to scan.
2. Fix some bugs.

xxconn commented:
Modified database.php and INFO.
